datasets/ONS_CPA/main.py

```diff
@@ -113,27 +108,21 @@ def product2cpa(p):
 cdids['Flow Directions'].cat.categories = cdids['Flow Directions'].cat.categories.map(
     lambda x: 'balance' if x == 'BAL' else 'imports' if x == 'IM' else 'exports' if x == 'EX' else None
 )
-# %%
-
-merged = pd.merge(df, cdids, on='cdid', how='left').drop_duplicates()
-merged.rename(columns={'cdid' : 'CDID'}, inplace=True)
-merged['Measure Type'] = "trade-in-goods"
-merged['Unit'] = "gbp-million"
-merged = merged[['ONS Partner Geography', 'Period', 'CDID', 'International Trade Basis',
-                             'Flow Directions', 'CPA 2008', 'Price Classification', 'Seasonal Adjustment','Value', 'Measure Type', 'Unit']].drop_duplicates()
+# -
+df = pd.merge(df, cdids, on='cdid', how='left').drop_duplicates()
+df.rename(columns={'cdid' : 'CDID'}, inplace=True)
+df['Measure Type'] = "trade-in-goods"
+df['Unit'] = "gbp-million"
+df = df[['ONS Partner Geography', 'Period', 'CDID', 'International Trade Basis',
+        'Flow Directions', 'CPA 2008', 'Price Classification', 'Seasonal Adjustment','Value', 'Measure Type', 'Unit']].drop_duplicates()
```

Comment:
'Measure Type' and 'Unit' could be deleted while dropping duplicates due to the previous change but before dropping duplicates, we need to validate the DataFrame for absence of duplicates and if duplicates are found then investigation needs to be done to decide whether duplicates needs to be dropped or is it genuine duplicates which shouldn't be dropped.